### ISSUE TITLE
It looks like you can't set the package last modified date in package_update

### ISSUE THREAD

**Issue by ffknob** [ckan/ckan#4687]
It looks like you can't set the package last modified date in package_update. What I am aiming to do is to update its resources with the modified date of the files in the filesystem, and then also modify the package's date to the date of the most recently changed file/resource...

But even though I set the "metadata_modified" in the context of package_update endpoitn, it seens to always set it as "now".

Is there any other way?

Thank you

https://github.com/ckan/ckan/blob/793f57fc19ae0d1b3c38319118589f08716f1a08/ckan/logic/action/update.py#L312

**Comment by amercader** [comment]
@ffknob you are right, this should be updated to check if `metadata_modified` is present in the data_dict and use that if so (defaulting to `now()` if not). This should be easy to implement in case you want to submit a PR.

Related, separate issues: 

* Only set metadata_modified if there is actually a change: https://github.com/ckan/ckan/pull/3535
* Better handling of this field: https://github.com/ckan/ckan/pull/3476

**Comment by wardi** [comment]
just need to be careful to remove metadata_modified from the data passed to package_update from resource_update and package_patch so that the date is still updated by these calls

**Comment by ffknob** [comment]
PR: https://github.com/ckan/ckan/pull/4697

**Comment by ffknob** [comment]
Sorry, I'm kind of new to the "contribution scene", but I really want to help this (and other) projects...

I'm trying to...

1) Figure it out why my commit is not passing the circle CI stage. What did I do wrong?

2) How could I run these testing cases locally, before a commit it to the repository? Is there a page with instructions?

Thank you!

**Comment by ffknob** [comment]
Some hypothesis:
- Did the package get saved into CKAN? (in order for "package_show" to work)
- Was the fake resource correctly linked to the fake dataset?


ckan/tests/logic/action/test_update.py:1089 => One of the two tests that are failing...
`def test_mimetype_by_upload_by_filename(self, mock_open):`

ckan/tests/logic/action/test_update.py:1120-1123 => It creates a fake resource
```
resource = factories.Resource(package=dataset,
                          url='http://localhost',
                          name='Test',
                          upload=test_resource)
```

ckan/tests/factories.py: => It creates the Resource based on the model.Resource class and I suppose it links is to the fake Dataset created and passed as a parameter
(creates a Resource based on ckan.model.Resource, which generates an id with _types.make_uuid())

(by this point I'm not sure if the resource has been sent to CKAN or not)

ckan/tests/logic/action/test_update.py:1135-1138 => now it tries to execute the update for the fake resource
```
res_update = helpers.call_action('resource_update',
                             id=resource['id'],
                             url='http://localhost',
                             upload=update_resource)
```

ckan/logic/action/update.py:71 => It tries to find a resource...
`resource = model.Resource.get(id)`

ckan/model/resource.py:112-120 => It tries to query by id, and if not found (I really don't know if by this point the resource has been sent to CKAN) I think it returns a new instance 
```
def get(cls, reference):
'''Returns a resource object referenced by its name or id.'''
if not reference:
    return None

resource = meta.Session.query(cls).get(reference)
if resource is None:
    resource = cls.by_name(reference)
return resource
```

ckan/logic/action/update.py:82-91 => Asks CKAN for package information, but it ends raising the exception
```
package_id = resource.package.id
pkg_dict = _get_action('package_show')(dict(context, return_type='dict'),
{'id': package_id})

for n, p in enumerate(pkg_dict['resources']):
if p['id'] == id:
    break
else:
log.error('Could not find resource %s after all', id)
raise NotFound(_('Resource was not found.'))
```

**Comment by wardi** [comment]
fixed by #8421 : sysadmins may now set the metadata_modified to any value, and the metadata_modified is otherwise now only updated when some metadata field has changed (instead of every time any update API is used).
